source/core/slang-string-slice-pool.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SLANG_STRING_SLICE_POOL_H
+#define SLANG_STRING_SLICE_POOL_H
+
+#include "slang-string.h"
+
+#include "list.h"
+#include "slang-memory-arena.h"
+#include "dictionary.h"
+
+namespace Slang {
+
+class StringSlicePool
```

Comment:
At a high level, this type is a *lot* like `NamePool`. I wonder if we need both in practice.